sdk/python/kfp/compiler/compiler.py

```diff
@@ -57,14 +57,23 @@ def my_pipeline(a: int = 1, b: str = "default value"):
 
   def __init__(
       self,
-      mode: dsl.PipelineExecutionMode = dsl.PipelineExecutionMode.V1_LEGACY):
+      mode: dsl.PipelineExecutionMode = dsl.PipelineExecutionMode.V1_LEGACY,
+      launcher_image: Optional[str] = None):
```

Comment:
Should this be the property the pipeline (`@dsl.pipeline(launcher_image=...)`)? Or maybe an attribute of the `kfp.dsl.PipelineConf` which is passed to `compiler.compile(pipeline_conf=...)`?

Reply:
No, I'd prefer this to be a property of the class itself for now.

Reply:
I also wonder whether this should be per-task.
Does it make sense for different component tasks to use different launchers?

Reply:
Not right now.